### PR TITLE
Rework torch usage for NPCs

### PR DESCRIPTION
**Pull request by akortunov**
Fixes [bug #4334](https://bugs.openmw.org/issues/4334).

Summary of changes:
1. NPC will hide light sources during bad weather, as they do in vanilla game.
The only difference here - in vanilla this feature does not work well during weather transitions, so I decided to check if there are precipitation particles in scene (rain, snow, etc.).
2. Since we do not want to unequip shields/torches when using two-handed weapons, there is no point to keep such limitation for NPCs. They will not unequp their weapons when they equip torches at night.

### CODE REVIEW

File: apps/openmw/mwworld/weather.cpp [hunk 1]
```diff
@@ -660,6 +660,10 @@ void WeatherManager::update(float duration, bool paused)
     mWindSpeed = mResult.mWindSpeed;
     mIsStorm = mResult.mIsStorm;
 
+    // For some reason Ash Storm is not considered as a precipitation weather in game
```
Comment (Capostrophic) [inline apps/openmw/mwworld/weather.cpp]: Well, there's not much difference between a blight storm and an ash storm mechanically, so why even have a special case for ash storms - merely for replicating the original perhaps buggy behavior?

Comment (akortunov) [reply apps/openmw/mwworld/weather.cpp]: > merely for replicating the original perhaps buggy behavior?

Yes. But I do not know if it is buggy or intended behaviour, so I decided to replicate it for now.
If we do not want this case, we just can remove it.

**Comment by scrawl** [comment]
Given that the only use of ```isDark()``` and ```isPrecipitation()``` is for checking if torches are required, can you consolidate them into one function?

**Comment by akortunov** [comment]
> Given that the only use of isDark() and isPrecipitation() is for checking if torches are required, can you consolidate them into one function?

Sure.

**Comment by scrawl** [comment]
Ok that's better. Just a minor suggestion for WeatherManager::useTorches. You could reduce the amount of typing and dependencies a bit. exterior/quasiExterior can be checked in World::useTorches (has nothing to do with the weather, anyway). Then get the timestamp in World, and change the other function to ```WeatherManager::useTorches(float hour)```. And voila, the WeatherManager won't need to know anything about World, at least not in this instance.

**Comment by scrawl** [comment]
```mPrecipitation``` should be initialized in the constructor.

**Comment by akortunov** [comment]
> And voila, the WeatherManager won't need to know anything about World

Weather manager uses the World at least in the playerTeleported() and update(float duration, bool paused). Alter these methods too?

**Comment by scrawl** [comment]
Ok sure, let's do that while we're at it.

**Comment by akortunov** [comment]
Done. A lot of changes to eliminate one line...

**Pull request merged**